packages/react-server/src/ReactFizzServer.js

```diff
@@ -1640,6 +1709,10 @@ function renderNodeDestructiveImpl(
   // something suspends.
   task.node = node;
   task.childIndex = childIndex;
+  // TODO: Should we pass keyPath as an argument and set that here, instead of
+  // setting it in the caller? It would reduce repetition, but it would be also
+  // be a redundant assignment in some cases, because not all node types
+  // contribute to the key path.
```

Comment:
@sebmarkbage Any thoughts?

Reply:
I think we shouldn't pass it as an argument. Not all paths set the formatContext or legacyContext etc neither. Only the ones that do set it (and reset it, unless there's an error).

Maybe childIndex also is not quite necessary because not all paths change the childIndex.